plugin/storage/integration/integration_test.go

```diff
@@ -0,0 +1,474 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/uber/jaeger/model"
+	"github.com/uber/jaeger/storage/spanstore"
+)
+
+type QueryType int
+
+type StorageIntegration struct {
+	ctx     context.Context
+	logger  *zap.Logger
+	writer  spanstore.Writer
+	reader  spanstore.Reader
+	cleanUp func() error
+	refresh func() error // function called between set-up and queries in each test
+}
+
+const (
+	byService QueryType = iota
+	bysNoN
+	bysNTags
+	bysNDuration
+	byNumOfTraces
+	// Add more queryTypes here
```

Comment:
you're right. I think I used this earlier when I needed the enums, but i changed the format and we no longer need it. i'll refactor